boot/zephyr/serial_adapter.c

```diff
@@ -192,7 +195,11 @@ static int
 boot_uart_fifo_init(void)
 {
 #ifdef CONFIG_BOOT_SERIAL_UART
+#if DT_HAS_CHOSEN(zephyr_uart_mcumgr)
+	uart_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_uart_mcumgr));
+#else
 	uart_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
+#endif
 #elif CONFIG_BOOT_SERIAL_CDC_ACM
 	uart_dev = DEVICE_DT_GET_ONE(zephyr_cdc_acm_uart);
```

Comment:
Currently, it is not really feasible to use a composite cdc acm usb device, such that one interface is used for logging and the other is used for serial recovery. This is because [`DEVICE_DT_GET_ONE`](https://docs.zephyrproject.org/2.7.0/reference/drivers/index.html?highlight=device_dt_get_one#c.DEVICE_DT_GET_ONE) returns an arbitrary valid node, which can of course collide with the cdc acm uart being used as the logging backend. 

So, maybe it would be useful to also try to use `zephyr,uart-mcumgr` as the default cdc acm uart device selection, and fall back to grabbing any valid node if it is not present?

```suggestion
#if DT_HAS_CHOSEN(zephyr_uart_mcumgr)
        uart_dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_uart_mcumgr));
#else
        uart_dev = DEVICE_DT_GET_ONE(zephyr_cdc_acm_uart);
#endif
```

Reply:
+1 for binding virtual port using zephyr,uart-mcumgr chosen node as well

Reply:
I would rather have that as another PR, because I am not that familiar with CDC ACM devices and reluctant to bring change I do not completely understand.